### ISSUE TITLE
Code Lens for autocompleting inferred type suggestions

### ISSUE THREAD

**Issue by kostmo** [swarm-game/swarm#980]
Just like the Haskell Language Server displays Code Lens entries that one can click to insert type signatures, we can do the same for swarm-lang.

![image](https://user-images.githubusercontent.com/261693/211176979-d9fd7896-4a0e-4117-b96b-bdc71731f6b5.png)

**Comment by xsebek** [comment]
AFAIK the inference functions only store the types of top-level bindings. (See #986)

@byorgey is there a way to get the types of all variables at definition sites?

**Comment by byorgey** [comment]
> is there a way to get the types of all variables at definition sites?

No, but there will be once I finish the work on the `annotate-ast` branch.
